internal/crypto/cert.go

```diff
@@ -59,3 +65,48 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// CompleteCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func CompleteCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
```

Comment:
From the implementation logic, looks like `MergeCertificateChain` is a better name.

Reply:
Updated.